reproman/support/jobs/orchestrators.py

```diff
@@ -479,6 +479,28 @@ def _datalad_format_command(ds, spec):
     spec["_extra_inputs_array"] = [exinputs] * len(batch_parameters)
 
 
+def call_check_dl_results(fn, failure_msg, *args, **kwds):
+    """Call function, checking status of DataLad-style results.
+
+    Parameters
+    ----------
+    fn : callable
+        Function that yields results.
+    failure_msg : str
+        Message to show on failure. The result dict is appended to this text.
+    *args, **kwds
+        Arguments passed to `fn`.
+
+    Raises
+    ------
+    An OrchestratorError if a failure is encountered.
+    """
+    for res in fn(*args, **kwds):
+        lgr.debug("datalad publish result: %s", res)
+        if res["status"] in ["error", "impossible"]:
```

Comment:
Wouldn't it better to instead check if it is not in ok status?
```suggestion
        if res["status"] not in ["ok", "notneeded"]:

```
ATM it should be the same effect
Ref: http://docs.datalad.org/en/stable/customization.html?highlight=Error%20impossible#expected-behavior
but if a new status is introduced - we don't know now if it would be okish or errorish, so wouldn't it be safer to assume it to be errorish?